### PR TITLE
Typecheck vs type inference tests

### PR DESCRIPTION
**Pull request by singpolyma**
This includes the unit tests from #440 and the "fix" from #439

### CODE REVIEW

**Comment by singpolyma** [comment]
Unresolved thread from previous PR: https://github.com/dhall-lang/dhall-lang/pull/440#issuecomment-476002112

File: tests/README.md [hunk 1]
```diff
@@ -99,6 +99,14 @@ Where `A` and `B` are:
 
 (Note: for the `failure` tests we don't build an Annotation, but just typecheck)
 
+### Running `type-inference` tests
+
+The tests should:
+- parse `A` and `B`
+- eventually resolve the imports in both (if not running `simple` or `unit` tests)
```
Comment (f-f) [inline tests/README.md]: We should take away this line so running these tests would have even less requirements, as no tests in this section require imports
We could also move here the `simple` tests (since they are basically unit tests too)

File: tests/type-inference/success/unit/RecursiveRecordMergeRecursivelyKindsA.dhall [hunk 2]
```diff
@@ -0,0 +1 @@
+{ x = { a = Type } } ∧ { x = { b = Type } }
```
Comment (f-f) [inline tests/type-inference/success/unit/RecursiveRecordMergeRecursivelyKindsA.dhall]: Could we move to another PR the tests that involve the discussion from https://github.com/dhall-lang/dhall-lang/pull/440#issuecomment-476002112?
So that we can get this merged and we have a place to contain all the discussion about it

The tests involved in this are:
- `RecursiveRecordTypeMergeRecursivelyKinds`
- `RecursiveRecordMergeRecursivelyKinds`
